### PR TITLE
Rover: add MIS_DONE_BEHAVE parameter

### PR DESCRIPTION
**Pull request by tajisoft**
MIS_DONE_BEHAVE parameter allow user to choose a mode when mission done.

### CODE REVIEW

**Comment by rmackay9** [comment]
Hi there.  One small change that's required is the commit should start with "Rover:"

**Comment by tajitaji551** [comment]
@rmackay9 I fixed the commit comment.

**Comment by tajisoft** [comment]
@rmackay9 I replied from another GitHub account. tajitaji 551 is also my account.

**Comment by rmackay9** [comment]
Ah, one more small issue.  The set_mode() call may fail so we should handle this.  It should probably try to set_mode() to Hold if it fails to enter loiter.

**Comment by tajisoft** [comment]
@rmackay9 Thank you for feedback. I modified to change to HOLD when set_mode returns false.

File: APMrover2/Parameters.cpp [hunk 1]
```diff
@@ -663,6 +663,13 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @Path: ../libraries/AP_Airspeed/AP_Airspeed.cpp
     AP_SUBGROUPINFO(airspeed, "ARSPD", 37, ParametersG2, AP_Airspeed),
 
+    // @Param: MIS_DONE_BEHAVE
+    // @DisplayName: Mission done behave
+    // @Description: Mode to become after mission done
+    // @Values: 0:Hold,1:Loiter
+    // @User: Standard
+    AP_GROUPINFO("MIS_DONE_BEHAVE", 38, ParametersG2, mis_done_behave, 0),
```
Comment (muramura) [inline APMrover2/Parameters.cpp]: I think that it is better to use the enum " MIS_DONE_BEHAVE_HOLD" definition.

File: APMrover2/commands_logic.cpp [hunk 2]
```diff
@@ -104,7 +104,14 @@ bool Rover::start_command(const AP_Mission::Mission_Command& cmd)
 // exit_mission - callback function called from ap-mission when the mission has completed
 void Rover::exit_mission()
 {
+    // play a tone
+    AP_Notify::events.mission_complete = 1;
+    // send message
     gcs().send_text(MAV_SEVERITY_NOTICE, "Mission Complete");
+
+    if (g2.mis_done_behave == MIS_DONE_BEHAVE_LOITER && set_mode(mode_loiter, MODE_REASON_MISSION_END)) {
```
Comment (muramura) [inline APMrover2/commands_logic.cpp]: If the current mode is the hold mode and BEHAVE is specified as Reuta, is it a recognition that "Switch from hold mode to Reuta mode"?

File: APMrover2/Parameters.cpp (hunk 1 above)
Comment (tajitaji551) [reply APMrover2/Parameters.cpp]: Thank you for your feedback. I think it is enough that devloper could see its value indecated in "@Values" section.

File: APMrover2/commands_logic.cpp (hunk 2 above)
Comment (tajisoft) [reply APMrover2/commands_logic.cpp]: I think the answer is YES. It would be LOITER after AUTO mission done.

**Comment by rmackay9** [comment]
squashed and merged, thanks!

**Pull request closed without merging**